src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ArrayBuffer.cs

```diff
@@ -39,16 +39,17 @@ public ArrayBuffer(int initialSize, bool usePool = false)
 
         public void Dispose()
         {
+            _activeStart = 0;
+            _availableStart = 0;
+
             if (_usePool)
             {
-                _activeStart = _availableStart = 0;
-
                 byte[] array = _bytes;
                 _bytes = null;
 
                 if (array != null)
                 {
-                    ArrayPool<byte>.Shared.Return(array);
+                    ArrayPool<byte>.Shared.Return(array, true);
```

Comment:
Why is clearing necessary?  Have you measured to see if there's any negative perf impact on HTTP/2?

Nit: `clearArray: true`

Reply:
There's a chance that user code takes an array from the shared pool and ends up barfing e.g. an auth token somewhere.

Reply:
Yeah this seems like a huge change. I wouldn't merge this until we know how it affects HTTP/1 and HTTP/2 performance. How frequently do we dispose ArrayBuffers? Multiple times per request?

Reply:
An alternative is to use a private pool, but that wouldn't respect GC pressure.

Reply:
To be clear: I think the important question is are we okay with that risk -- if not, perf impact doesn't matter, we must do something different (e.g. clear buffers or use private pool).

CC @GrabYourPitchforks @bartonjs